File: src/main/kotlin/com/refah/walletwrapper/utils/ReadExcelData.java
package com.refah.walletwrapper.utils;

import org.apache.poi.ss.usermodel.Cell;
import org.apache.poi.ss.usermodel.Row;
import org.apache.poi.xssf.usermodel.XSSFSheet;
import org.apache.poi.xssf.usermodel.XSSFWorkbook;
import org.springframework.web.multipart.MultipartFile;

import java.io.InputStream;
import java.util.ArrayList;
import java.util.Iterator;

public class ReadExcelData {

    public static ArrayList<ArrayList<String>> importExcelData(MultipartFile input) {
        ArrayList<ArrayList<String>> excelData = new ArrayList<>();
        try {
            InputStream file = input.getInputStream();
            XSSFWorkbook workbook = new XSSFWorkbook(file);
            XSSFSheet sheet = workbook.getSheetAt(0);
            Iterator<Row> rowIterator = sheet.iterator();
            while (rowIterator.hasNext()) {
                Row row = rowIterator.next();
                Iterator<Cell> cellIterator = row.cellIterator();
                ArrayList<String> rowDetail = new ArrayList<>();
                while (cellIterator.hasNext()) {
                    Cell cell = cellIterator.next();
                    switch (cell.getCellType()) {
                        case NUMERIC:
                            rowDetail.add(new Double(cell.getNumericCellValue()).longValue() + "");
                            break;
                        case STRING:
                            rowDetail.add(cell.getStringCellValue());
                            break;
                    }
                }
                excelData.add(rowDetail);
            }
            file.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return excelData;
    }
}

File: src/main/kotlin/com/refah/walletwrapper/utils/ReadExcelData.kt
package com.refah.walletwrapper.utils

import org.apache.poi.ss.usermodel.CellType
import org.apache.poi.ss.usermodel.Row
import org.apache.poi.xssf.usermodel.XSSFSheet
import org.apache.poi.xssf.usermodel.XSSFWorkbook
import org.springframework.web.multipart.MultipartFile


class ReadExcelData {

    companion object {
        fun importExcelData( file2 : MultipartFile): ArrayList<ArrayList<String>> {
            val excelData = ArrayList<ArrayList<String>>()
            try {
                val file = file2.inputStream
                val workbook = XSSFWorkbook(file)
                val sheet: XSSFSheet = workbook.getSheetAt(0)
                val rowIterator: Iterator<Row> = sheet.iterator()

                while (rowIterator.hasNext()) {
                    val row: Row = rowIterator.next()
                    val cellIterator = row.cellIterator()
                    val rowDetail = ArrayList<String>()
                    while (cellIterator.hasNext()) {
                        val cell = cellIterator.next()
                        when (cell.cellType) {
                            CellType.NUMERIC -> rowDetail.add(cell.numericCellValue.toInt().toString() + "")
                            CellType.STRING -> rowDetail.add(cell.stringCellValue)
                        }
                    }
                    excelData.add(rowDetail)
                }
                file.close()
            } catch (e: Exception) {
                e.printStackTrace()
            }
            return excelData
        }
    }
}